internal/cmd/agent/controller.go

```diff
@@ -0,0 +1,105 @@
+package agent
+
+import (
+	"context"
+	"time"
+
+	"github.com/rancher/fleet/internal/cmd/agent/controllers"
+	"github.com/rancher/fleet/internal/cmd/agent/register"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/rancher/lasso/pkg/mapper"
+	"github.com/rancher/wrangler/v2/pkg/kubeconfig"
+	"github.com/rancher/wrangler/v2/pkg/ratelimit"
+	"github.com/rancher/wrangler/v2/pkg/ticker"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/discovery/cached/memory"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/restmapper"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+// defaultNamespace is the namespace to use for resources that don't specify a namespace, e.g. "default"
+const defaultNamespace = "default"
+
+// start the fleet agent
+func start(ctx context.Context, kubeConfig, namespace, agentScope string) error {
+	clientConfig := kubeconfig.GetNonInteractiveClientConfig(kubeConfig)
+	kc, err := clientConfig.ClientConfig()
+	if err != nil {
+		return err
+	}
+
+	agentInfo, err := register.Get(ctx, namespace, kc)
+	if err != nil {
+		logrus.Fatal(err)
```

Comment:
Yes, all these logrus calls should be removed in the future https://github.com/rancher/fleet/pull/1772/files#diff-a08a0ecbbab37eaea426b6b8bc55a0d5107c75cac3db92153ef7d3329dc47650R79. But the second point you make is interesting. Why do we exit instead of return...
I'll change it in the next PR.